File: mobile/src/main/java/com/siliconlabs/bledemo/Bluetooth/BLE/Values/ByteArrayValue.java
package com.siliconlabs.bledemo.Bluetooth.BLE.Values;

import android.bluetooth.BluetoothGattCharacteristic;

/**
 * The plain byte array of the characteristic.
 */
public class ByteArrayValue {
    public static class Factory implements ValueFactory<byte[]> {
        @Override
        public byte[] create(BluetoothGattCharacteristic value) {
            return value.getValue();
        }
    }
}

File: mobile/src/main/java/com/siliconlabs/bledemo/Bluetooth/BLE/Values/ByteArrayValue.kt
package com.siliconlabs.bledemo.bluetooth.ble.values

import android.bluetooth.BluetoothGattCharacteristic

/**
 * The plain byte array of the characteristic.
 */
class ByteArrayValue {
    class Factory : ValueFactory<ByteArray> {
        override fun create(value: BluetoothGattCharacteristic): ByteArray {
            return value.value
        }
    }
}